sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -1,20 +1,742 @@
+      }
+      log(SentryLevel.INFO, "Not possible to read external files directory");
+      return null;
+    }
+    log(SentryLevel.INFO, "External storage is not mounted or emulated.");
+    return null;
+  }
+
+  @SuppressWarnings("deprecation")
+  private File getExternalStorageDep() {
+    return Environment.getExternalStorageDirectory();
+  }
+
+  /**
+   * Get the total amount of external storage, in bytes, or null if no external storage is mounted.
+   *
+   * @return the total amount of external storage, in bytes, or null if no external storage is
+   *     mounted
+   */
+  private Long getTotalExternalStorage(StatFs stat) {
+    try {
+      long blockSize = getBlockSizeLong(stat);
+      long totalBlocks = getBlockCountLong(stat);
+      return totalBlocks * blockSize;
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting total external storage amount.", e);
+      return null;
+    }
+  }
+
+  private boolean isExternalStorageMounted() {
+    return Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)
+        && !Environment.isExternalStorageEmulated();
+  }
+
+  /**
+   * Get the unused amount of external storage, in bytes, or null if no external storage is mounted.
+   *
+   * @return the unused amount of external storage, in bytes, or null if no external storage is
+   *     mounted
+   */
+  private Long getUnusedExternalStorage(StatFs stat) {
+    try {
+      long blockSize = getBlockSizeLong(stat);
+      long availableBlocks = getAvailableBlocksLong(stat);
+      return availableBlocks * blockSize;
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting unused external storage amount.", e);
+      return null;
+    }
+  }
+
+  /**
+   * Get the DisplayMetrics object for the current application.
+   *
+   * @return the DisplayMetrics object for the current application
+   */
+  private DisplayMetrics getDisplayMetrics() {
+    try {
+      return context.getResources().getDisplayMetrics();
+    } catch (Exception e) {
+      log(SentryLevel.ERROR, "Error getting DisplayMetrics.", e);
+      return null;
+    }
+  }
+
+  private OperatingSystem getOperatingSystem() {
+    OperatingSystem os = new OperatingSystem();
+    os.setName("Android");
+    os.setVersion(Build.VERSION.RELEASE);
+    os.setBuild(Build.DISPLAY);
+    os.setKernelVersion(getKernelVersion());
+    os.setRooted(isRooted());
+    //    os.setRawDescription("");  whats that? maybe the const of VERSION_CODES?
+
+    return os;
+  }
+
+  private App getApp(PackageInfo packageInfo) {
+    App app = new App();
+    app.setIdentifier(packageInfo.packageName);
+    app.setStartTime(
+        DateUtils
+            .getCurrentDateTime()); // is it App. start? this was not supposed to be a lazy class?
+    //    app.setHash(); whats that?
+    //    app.setBuildType(); whats that? might be possible with BuildConfig.BUILD_VARIANT but Apps
+    // side, also for flavor
```

Comment:
Should we expand the protocol to be more Android friendly then?
Add a Flavor and variant field or something?

Reply:
let me check if it's even possible on the lib. side, these properties with BuildConfig might be available only for the App. so the user will need to set them on BeforeSend

Reply:
users should set that on BeforeSend if they want to.
expanding the protocol might be the case.